### PR TITLE
EDO-601. Add the mapper admin endpoints

### PR DESCRIPTION
**Pull request by oleg1std**
Add endpoints to manage the mapper from the admin panel.
- Deactivate accommodations
- Combine accommodations
- Remove a supplier from an accommodation

### CODE REVIEW

File: Api/Controllers/AdministratorControllers/MapperManagementControllers/AccommodationController.cs [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+            var (_, isFailure, _, error) = await _mapperManagementClient.DeactivateAccommodations(request, DeactivationReasons.WrongMatching, cancellationToken);
+
+            if (isFailure)
+                return BadRequest(error);
+
+            return NoContent();
+        }
+
+
+        [HttpPost("{htAccommodationId}/suppliers/{supplier}/accommodations/{supplierId}/remove")]
```
Comment (kirillta) [inline Api/Controllers/AdministratorControllers/MapperManagementControllers/AccommodationController.cs:68]: In result we have `api/{v:apiVersion}/admin/mapper/management/accommodations/{htAccommodationId}/suppliers/{supplier}/accommodations/{supplierId}/remove`

File: Api/Infrastructure/Logging/LogEvents.json [hunk 2]
```diff
@@ -76,5 +77,7 @@
...
     {"id": 1802, "name": "MapperClientUnexpectedResponse", "level": "Error", "template":  "Unexpected response received from mapper. StatusCode: `{StatusCode}`, request uri: `{Uri}`, response: {Response}", "argumentTypes": [ "System.Net.HttpStatusCode", "System.Uri", "string" ]},
-    {"id": 1803, "name": "MapperClientRequestTimeout", "level": "Warning", "template":  "Request to mapper failed with timeout"}
+    {"id": 1803, "name": "MapperClientRequestTimeout", "level": "Warning", "template":  "Request to mapper failed with timeout"},
+    {"id": 1804, "name": "MapperManagementClientUnexpectedResponse", "level": "Error", "template":  "Unexpected response received from a mapper management endpoint. StatusCode: `{StatusCode}`, request uri: `{Uri}`, response: {Response}", "argumentTypes": [ "System.Net.HttpStatusCode", "System.Uri", "string" ]},
+    {"id": 1806, "name": "MapperManagementClientRequestTimeout", "level": "Warning", "template":  "Request to a mapper management endpoint failed with timeout"}
```
Comment (kirillta) [inline Api/Infrastructure/Logging/LogEvents.json:82]: Could these templates be more descriptive?

File: Api/Controllers/AdministratorControllers/MapperManagementControllers/AccommodationController.cs (hunk 1 above)
Comment (oleg1std) [reply Api/Controllers/AdministratorControllers/MapperManagementControllers/AccommodationController.cs:68]: The path and arguments have been taken from the mapper.
The path in the mapper is : _api/{v:apiVersion}/AccommodationsManagement/accommodations/{htAccommodationId}/suppliers/{supplier}/accommodations/{supplierCode}/remove_
I think @hasmikharutyunyan will be unhappy if we change the path

File: Api/Infrastructure/Logging/LogEvents.json (hunk 2 above)
Comment (oleg1std) [reply Api/Infrastructure/Logging/LogEvents.json:82]: These events are for the notification of some unexpected errors. They were added by the analogy with events from the AccommodationMapperClient

File: Api/Controllers/AdministratorControllers/MapperManagementControllers/AccommodationController.cs (hunk 1 above)
Comment (kirillta) [reply Api/Controllers/AdministratorControllers/MapperManagementControllers/AccommodationController.cs:68]: But we can't leave them that way either

Comment (mingazhev) [reply Api/Controllers/AdministratorControllers/MapperManagementControllers/AccommodationController.cs:68]: Ok for me but should not we put `supplier` and `supplierId` to the post request body?

File: Api/Controllers/AdministratorControllers/MapperManagementControllers/AccommodationController.cs [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+        /// Deactivates accommodations by htId with the reason is invalid mapping 
+        /// </summary>
+        /// <param name="request"></param>
+        /// <param name="cancellationToken"></param>
+        /// <returns></returns>
+        [HttpPost("deactivate")]
+        [ProducesResponseType(StatusCodes.Status200OK)]
+        [ProducesResponseType(typeof(ProblemDetails), StatusCodes.Status400BadRequest)]
+        [AdministratorPermissions(AdministratorPermissions.MapperAccommodationManagement)]
+        public async Task<IActionResult> DeactivateBecauseOfInvalidMatching([FromBody] AccommodationsRequest request, CancellationToken cancellationToken = default)
```
Comment (mingazhev) [inline Api/Controllers/AdministratorControllers/MapperManagementControllers/AccommodationController.cs:57]: Should not it be DeactivateAccommodationRequest?

File: Api/AdministratorServices/Models/Mapper/DeactivationReasons.cs [hunk 4]
```diff
@@ -0,0 +1,13 @@
+﻿namespace HappyTravel.Edo.Api.AdministratorServices.Models.Mapper
+{
+    public enum DeactivationReasons
+    {
+        None = 0,
```
Comment (mingazhev) [inline Api/AdministratorServices/Models/Mapper/DeactivationReasons.cs:5]: Can we add this to MapperContracts?

**Review by kirillta — APPROVED** [review]

File: Api/Controllers/AdministratorControllers/MapperManagementControllers/AccommodationController.cs (hunk 1 above)
Comment (oleg1std) [reply Api/Controllers/AdministratorControllers/MapperManagementControllers/AccommodationController.cs]: fixed

File: Api/Controllers/AdministratorControllers/MapperManagementControllers/AccommodationController.cs (hunk 3 above)
Comment (oleg1std) [reply Api/Controllers/AdministratorControllers/MapperManagementControllers/AccommodationController.cs]: fixed

File: Api/AdministratorServices/Models/Mapper/DeactivationReasons.cs (hunk 4 above)
Comment (oleg1std) [reply Api/AdministratorServices/Models/Mapper/DeactivationReasons.cs]: fixed

File: Api/HappyTravel.Edo.Api.csproj [hunk 5]
```diff
@@ -72,4 +72,7 @@
     </None>
     <None Remove=".appsettings.Local.json.swp" />
   </ItemGroup>
+  <ItemGroup>
+    <Folder Include="AdministratorServices\Mapper" />
```
Comment (mingazhev) [inline Api/HappyTravel.Edo.Api.csproj:76]: Is this needed?

File: Api/Controllers/AdministratorControllers/MapperManagementControllers/AccommodationController.cs [hunk 6]
```diff
@@ -0,0 +1,84 @@
...
+
+        [HttpPost("{htAccommodationId}/suppliers/remove")]
+        [ProducesResponseType(StatusCodes.Status200OK)]
+        [ProducesResponseType(typeof(ProblemDetails), StatusCodes.Status400BadRequest)]
+        [AdministratorPermissions(AdministratorPermissions.MapperAccommodationManagement)]
+        public async Task<IActionResult> RemoveSupplier([FromRoute] string htAccommodationId, [FromBody] RemoveSupplierRequest request, CancellationToken cancellationToken = default)
+        {
+            var (_, isFailure, _, error) = await _mapperManagementClient.RemoveSupplier(htAccommodationId, request, cancellationToken);
+            if (isFailure)
+                return BadRequest(error);
```
Comment (mingazhev) [inline Api/Controllers/AdministratorControllers/MapperManagementControllers/AccommodationController.cs:76]: BadRequestWithProblemDetails I guess

**Review by mingazhev — APPROVED** [review]

File: Api/Controllers/AdministratorControllers/MapperManagementControllers/AccommodationController.cs (hunk 6 above)
Comment (oleg1std) [reply Api/Controllers/AdministratorControllers/MapperManagementControllers/AccommodationController.cs]: fixed

File: Api/HappyTravel.Edo.Api.csproj (hunk 5 above)
Comment (oleg1std) [reply Api/HappyTravel.Edo.Api.csproj]: fixed

**Review by kirillta — APPROVED** [review]

**Pull request merged**